File: leetcode/by_level/easy/js/944.js
const solution = (arr) => {
  let D = 0;
  for (let i = 0; i < arr[0].length; i++) {
    let tempArr = []
    let decreased = 0
    for (let j = 0; j < arr.length; j++) {
      tempArr.push(arr[j].split('')[i])
    }
    for (let k = 0; k < tempArr.length - 1; k++) {
      if (tempArr[k + 1].charCodeAt() < tempArr[k].charCodeAt()) {
        decreased++
      }
    }
    if (decreased > 0) D++
  }
  return D
}

const input =
  ["cba", "daf", "ghi"]

console.log(solution(input))

File: leetcode/by_level/easy/py/944.py
def solution(A):
  if len(A) == 0:
    return 0
  D = 0

  for y in range(len(A[0])):
    for x in range(len(A) - 1):
      if A[x][y] > A[x+1][y]:
        D = D + 1
        break
  
  return D

input = ["cba","daf","ghi"]

print(solution(input))